src/frontend/src/components/EditTripsModal.tsx

```diff
@@ -1,32 +1,198 @@
+        const destinations = profileBuffer.destinations.map(d => {
+            return { ...d, primary: d.location.label === selection };
+        });
+        setProfileBuffer({ ...profileBuffer, destinations: destinations });
+    };
+
+    // Directly update networks state for traffic changes. Avoids
+    // race conditions correctly updating activeMode following user profile update
+    const onTrafficChange = (keys: Iterable<Key, void, undefined>) => {
+        const selection = [...keys][0];
+        if (!selection) return;
+        dispatch(setTrafficConditions(selection as TrafficType));
+    };
+
+    const onChangeHasVehicleToggle = (keys: Set<Key>) =>
+        setProfileBuffer({
+            ...profileBuffer,
+            hasVehicle: keys.has("car"),
+            useCommuterRail: keys.has("car")
+                ? false
+                : profileBuffer.useCommuterRail,
+        });
+
+    const onChangeCommuterCheckbox = (value: boolean) =>
+        setProfileBuffer({
+            ...profileBuffer,
+            useCommuterRail: value,
+        });
+
+    const onOpenChange = (isOpen: boolean) => {
+        dispatch(
+            updateUserProfile({
+                ...profileBuffer,
+                voucherRooms: profileBuffer.rooms,
+                importanceAccessibility: parseInt(
+                    profileBuffer.importanceAccessibility
+                ),
+                importanceSchools: parseInt(profileBuffer.importanceSchools),
+                importanceViolentCrime: parseInt(
+                    profileBuffer.importanceViolentCrime
+                ),
+            })
+        );
+        dispatch(setIsEditTripsOpen(isOpen));
+    };
+
     return (
         <ModalOverlay
             isDismissable
             isMobile
             isOpen={modalOpen}
-            onOpenChange={(isOpen: boolean) =>
-                dispatch(setIsEditTripsOpen(isOpen))
-            }
+            onOpenChange={onOpenChange}
         >
-            <Modal size="large">
-                <Dialog>
-                    <Heading slot="title">Trips</Heading>
-                    <ModalCloseButton
-                        onPress={() => dispatch(setIsEditTripsOpen(false))}
-                    />
-                    <div></div>
+            <Modal size="medium" className="p-5 max-h-full overflow-y-scroll">
+                <Dialog className="flex flex-col w-full gap-5">
+                    <div>
+                        <Heading slot="title" className={modalHeadingClassName}>
+                            {t("editTripsModal.dialogHeading")}
+                        </Heading>
+
+                        <p className="text-sm text-gray-600">
+                            {t("editTripsModal.dialogSubHeading")}
+                        </p>
+                        <ModalCloseButton onPress={() => onOpenChange(false)} />
+                    </div>
+                    <div className="flex flex-col gap-4">
+                        <Accordion
```

Comment:
I'd personally make these destination options as radio buttons that look like the vertical button group in the design, which would likely be closer to what it achieves semantically. You might have considered it too. But that would mean a new component to be created. Given the remaining time and budget, using an accordion works too.

Reply:
Yeah good to document this decision here, because I did think about that but given our budget/an already-styled accordion this felt like the best way forward. It'd be great to not only use that new radio group component here but also for the commute group component when a user's setup for car directions or a neighborhood is too far. I'll open up a tech debt issue for us and hopefully we can sneak it in at the end